easysid.cpp

```diff
@@ -152,42 +157,66 @@ EXPORT void set_filter_res(uint8_t res)
     sid.write(23, reg);
 }
 
-EXPORT void set_filter_enable(uint8_t voice)
+EXPORT void set_filter(uint8_t voice, bool enable)
 {
     if (voice < 1 || voice > 3) {
         return;
     }
     char reg = sid.read_state().sid_register[23];
-    reg &= ~0x0f;
-    reg |= 1 < (voice - 1);
+    reg &= ~1 << (voice - 1);
+    if (enable) {
+        reg |= (1 << voice - 1);
+    }
+    sid.write(23, reg);
+}
+
+EXPORT void set_filter_external(bool enable)
+{
+    char reg = sid.read_state().sid_register[23];
+    reg &= ~(1 << 3) & 0x0f;
```

Comment:
And as discussed, this is still wrong. You're only modifying a single bit here, it should look pretty much the same as the other functions that modify a single bit.